pkg/pool-manager/pool_test.go

```diff
@@ -414,4 +416,56 @@ var _ = Describe("Pool", func() {
 		})
 	})
 
+	Describe("Multus Network Annotations API Tests", func() {
+		Context("when pool-manager is configured with available addresses, Pod is created with networks annotation with mac-address", func() {
+			pod := v1.Pod{ObjectMeta: metav1.ObjectMeta{Name: "testPod", Namespace: "default"}}
+			poolManager := &PoolManager{}
+
+			BeforeEach(func() {
+				poolManager = createPoolManager("02:00:00:00:00:00", "02:00:00:00:00:02", &vmConfigMap)
+				Expect(poolManager).ToNot(Equal(nil), "should create pool-manager")
+			})
+
+			table.DescribeTable("should allocate mac-address correspond to the one specified in the networks annotation",
+				func(networkRequestAnnotation string) {
+					pod.Annotations = map[string]string{networksAnnotation: fmt.Sprintf("%s", networkRequestAnnotation)}
+
+					By("Request specific mac-address by adding the address to the networks pod annotation")
+					err := poolManager.AllocatePodMac(&pod)
+					Expect(err).ToNot(HaveOccurred(), "should allocate mac address and ip address correspond to networks annotation")
+
+					By("Convert obtained networks annotation JSON to multus.NetworkSelectionElement array")
+					obtainedNetworksAnnotationJson := pod.Annotations[networksAnnotation]
+					obtainedNetworksAnnotation := []multus.NetworkSelectionElement{}
+					err = json.Unmarshal([]byte(obtainedNetworksAnnotationJson), &obtainedNetworksAnnotation)
+					Expect(err).ToNot(HaveOccurred(), "should convert obtained annotation as json to multus.NetworkSelectionElement")
+
+					By("Convert expected networks annotation JSON to multus.NetworkSelectionElement array")
+					expectedNetworksAnnotation := []multus.NetworkSelectionElement{}
+					err = json.Unmarshal([]byte(networkRequestAnnotation), &expectedNetworksAnnotation)
+					Expect(err).ToNot(HaveOccurred(), "should convert expected annotation as json to multus.NetworkSelectionElement")
+
+					By("Compare between each obtained and expected network request")
+					for _, expectedNetwork := range expectedNetworksAnnotation {
+						Expect(obtainedNetworksAnnotation).To(ContainElement(expectedNetwork))
+					}
+				},
+				table.Entry("with single ip-address request as string array",
+				`[{"name":"ovs-conf","namespace":"default","ips":["10.10.0.1"],"mac":"02:00:00:00:00:00"}]`),
+				table.Entry("with multiple ip-address request as string array",
+				`[{"name":"ovs-conf","namespace":"default","ips":["10.10.0.1","10.10.0.2","10.0.0.3"],"mac":"02:00:00:00:00:00"}]`),
+				table.Entry("with multiple networks requsets",`[
+						{"name":"ovs-conf","namespace":"default","ips":["10.10.0.1","10.10.0.2","10.0.0.3"],"mac":"02:00:00:00:00:00"},
+						{"name":"cnv-bridge","namespace":"openshift-cnv","ips":["192.168.66.100","192.168.66.101"],"mac":"02:F0:F0:F0:F0:F0"}
+				]`),
+			)
+			It("should fail to allocate requested mac-address, with ip-address request as string instead of string array", func(){
```

Comment:
This is something that should be tested on Multus, but I don't see how it concerns us in KMP.

Reply:
This is how we check that the pool-manager will not handle request that multus wouldn't that eventually will fail

Reply:
If it will eventually fail on Multus, we should let it fail on Multus. There is a lot of other components that it may fail on later. We should not care about any of those. We should test for our failures and our positive flow. Duplicating tests of other components will leave us with unmaintainable code-base. Unless you convince me this brings any value to this project.

Reply:
This PR is about fixing this bug https://bugzilla.redhat.com/show_bug.cgi?id=1816971
and preventing another issue like that, so I added two cases where we use the new `ips` property and another case where `ips` is not valid.
This is about testing [parsePodNetworkAnnotation](https://github.com/k8snetworkplumbingwg/kubemacpool/blob/97e9201d0609d2f3ae7f23a68b4893f5961f2113/pkg/pool-manager/pod_pool.go#L286) which is where the error comes from, if you think it is redundant I will remove it.

Reply:
I don't see how adding a specific negative test helps us. But maybe I'm missing something, I'd like to hear from the other reviewers.

@RamLavi @qinqon

Reply:
Well testing the behaviour on bad format is not a bad thing, at least we know that we don't panic.

Reply:
Fair enough then

Reply:
If this test fail its indicate we are not aligned with the API of NetworkSelectionElement we parse and use